hazelcast/src/test/java/com/hazelcast/config/YamlConfigBuilderTest.java

```diff
@@ -3008,6 +3008,7 @@ public void testHotRestart() {
         HotRestartPersistenceConfig hotRestartPersistenceConfig = config.getHotRestartPersistenceConfig();
 
         assertTrue(hotRestartPersistenceConfig.isEnabled());
+        assertTrue(hotRestartPersistenceConfig.isAutoRemoveStaleData());
```

Comment:
We miss explicitly configuring `auto-remove-stale-data` in the YAML string above that demonstrates the issue with schema validation. This assertion passes even without the json schema fix because `true` is the default value of `autoRemoveStaleData`.

Reply:
ohh, ok, thanks a lot. it is fixed now